src/user/test/e2e/getUser.e2e.spec.ts

```diff
@@ -0,0 +1,53 @@
+import { ExecutionContext, HttpStatus, ValidationPipe } from '@nestjs/common';
+import { NestApplication } from '@nestjs/core';
+import { Test, TestingModule } from '@nestjs/testing';
+import request from 'supertest';
+
+import { UserInfoDto } from '../../dto/user.dto';
+import { UserController } from '../../user.controller';
+import { UserService } from '../../user.service';
+import { generateUser } from '../mock/user.model.mock';
+import { mockUserService } from '../mock/user.service.mock';
+
+describe('user', () => {
+  let service: UserService;
+  let app: NestApplication;
+  let mockUser: UserInfoDto;
+
+  beforeEach(async () => {
+    mockUser = generateUser();
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [],
+      controllers: [UserController],
+      providers: [
+        {
+          provide: UserService,
+          useValue: mockUserService,
+        },
+      ],
+    }).compile();
+    service = module.get(UserService);
+    app = module.createNestApplication();
+    app.useGlobalPipes(new ValidationPipe({ transform: true, transformOptions: { enableImplicitConversion: true } }));
+    app.useGlobalGuards({
+      canActivate(context) {
+        return true;
+      },
+    });
+    await app.init();
+  });
+  afterEach(async () => {
+    await app.close();
+  });
```

Comment:
I don't think we need to run this on each test, we can start the app before all tests and close it after all.